ApplicationLibCode/ReservoirDataModel/RigCaseCellResultsData.cpp

```diff
@@ -2002,6 +2011,72 @@ void RigCaseCellResultsData::computeIndexResults()
     }
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RigCaseCellResultsData::computeFaultDistance()
+{
+    size_t reservoirCellCount = activeCellInfo()->reservoirCellCount();
+    if ( reservoirCellCount == 0 ) return;
+
+    size_t resultIndex = findScalarResultIndexFromAddress(
+        RigEclipseResultAddress( RiaDefines::ResultCatType::STATIC_NATIVE, RiaResultNames::faultDistanceName() ) );
+
+    if ( resultIndex == cvf::UNDEFINED_SIZE_T ) return;
+
+    std::vector<std::vector<double>>& result = m_cellScalarResults[resultIndex];
+
+    if ( result.empty() ) result.resize( 1 );
+
+    bool shouldCompute = false;
+    if ( result[0].size() < reservoirCellCount )
+    {
+        result[0].resize( reservoirCellCount, std::numeric_limits<double>::infinity() );
+        shouldCompute = true;
+    }
+
+    if ( !shouldCompute ) return;
+
+    const std::vector<RigCell>& globalCellArray = m_ownerMainGrid->globalCellArray();
+
+    long long numCells = static_cast<long long>( globalCellArray.size() );
+
+    std::vector<cvf::StructGridInterface::FaceType> faceTypes = cvf::StructGridInterface::validFaceTypes();
+
+    // Preprocessing: create vector of all fault face centers.
```

Comment:
Suggestion add `RigActiveCellInfo::reservoirCells` that will return a vector with reservoir cells. Use this vector to loop through all cells. Then testing with `isActive` can be avoided.